content/en/istio-ca-certs-integrations/cert-manager-integration/_index.md

```diff
@@ -188,13 +184,170 @@ patches:
+```
+
+By default, Istio configures destination workloads using `PERMISSIVE` mode. This
+mode means a service can accept both plain text and mTLS connections. To ensure
+mTLS is being used between sleep and httpbin workloads, set the mode to `STRICT`
+for namespace "foo".
+```sh
+kubectl apply -n foo -f - <<EOF
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+  name: "default"
+spec:
+  mtls:
+    mode: STRICT
+EOF
+```
+
+Test mTLS from the sleep pod to the httpbin pod.
+```sh
+kubectl -n foo exec -it deploy/sleep -c sleep -- curl -o /dev/null -s -w '%{http_code}\n' http://httpbin.foo:8000/headers
+```
+You should receive an `200` response code.
+
+## Secure Gateways
+
+Istio Gateways support exposing secure HTTPS services using simple or mutual TLS.
+cert-manager can be used to generate the gateway's certificate and key. To get started,
+configure a Certificate that will be used to securely expose the httpbin workload.
+The Certificate must be in the same namespace as the associated ingress gateway
+deployment, e.g. `istio-system`.
+```sh
+kubectl apply -n istio-system -f - <<EOF
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: httpbin
+spec:
+  commonName: httpbin.example.com
+  dnsNames:
+  - httpbin.example.com
+  issuerRef:
+    group: cert-manager.io
+    kind: Issuer
+    name: istio-ca
+  secretName: httpbin-cert
+EOF
+```
+
+Verify the Certificate is reporting `READY=True`.
+```sh
+kubectl get cert/httpbin -n istio-system
+```
+
+Create a Gateway and VirtualService for the httpbin workload.
+```sh
+kubectl apply -n foo -f - <<EOF
+apiVersion: networking.istio.io/v1beta1
+kind: Gateway
+metadata:
+  name: httpbin
+spec:
+  servers:
+  - port:
+      number: 443
+      name: httpbin-tls
+      protocol: HTTPS
+    tls:
+      mode: SIMPLE
+      credentialName: httpbin-cert
+    hosts:
+    - httpbin.example.com
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: httpbin
+spec:
+  hosts:
+  - "httpbin.example.com"
+  gateways:
+  - httpbin
+  http:
+  - match:
+    - uri:
+        prefix: /
+    route:
+    - destination:
+        host: httpbin.foo.svc.cluster.local
+EOF
+```
+Mode `SIMPLE` secures connections with standard TLS semantics, i.e. the client
+authenticates the gateway's certificate.
+
+Copy the Istio CA certificate used by the client to establish a secure
+connection with the ingress gateway.
+```sh
+kubectl get cm/istio-ca-root-cert -n foo -o json | jq -r '.data."root-cert.pem"' > root-cert.pem
+```
+
+Before connecting to the httpbin service, determine your [ingress gateway's IP address][1].
+The `--cacert` flag is used to specify the Istio CA root certificate used to authenticate
+the ingress gateway's server certificate.
+```sh
+curl -o /dev/null -s -w '%{http_code}\n' --cacert --resolve "httpbin.example.com:443:$INGRESS_HOST" root-cert.pem "https://httpbin.example.com/status/200"
```

Comment:
[nit] `--connect` option is more universal than `--resolve`. It will support ingresses with a DNS name instead of IP address.

See https://daniel.haxx.se/blog/2018/04/05/curl-another-host/

Reply:
The arguments are in the wrong order - `root-cert.pem` should immediately follow `--cacert`.

Reply:
N/A, section removed.